src/array.jl

```diff
@@ -364,15 +364,16 @@ arraytype(A::CatArray...) = NullableCategoricalArray
 
 function vcat(A::CatArray...)
     newlevels, ordered = mergelevels(map(levels, A)...)
+    ordered &= any(isordered, A)
```

Comment:
OK. But then if `ordered` is false, you're doing the check unnecessarily (same for the one below). Better write `ordered = ordered && any(...) && all(...)`.